scene/gui/code_edit.cpp

```diff
@@ -2856,7 +2856,7 @@ void CodeEdit::_filter_code_completion_candidates_impl() {
 	const int caret_line = get_caret_line();
 	const int caret_column = get_caret_column();
 	const String line = get_line(caret_line);
-	ERR_FAIL_INDEX_MSG(caret_column - 1, line.length(), "Caret column exceeds line length.");
+	ERR_FAIL_INDEX_MSG(caret_column, line.length() + 1, "Caret column exceeds line length.");
```

Comment:
Are you sure about the `line.length() + 1`? Won't this lead to out of bound access when on the last column?

Reply:
Yes, for example in a line with 3 chars, the caret's column can be 0 (before all chars), 1, 2, or 3 (after all chars)